sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java

```diff
@@ -232,35 +259,37 @@ public HttpClient build() {
         return new OkHttpAsyncHttpClient(httpClientBuilder.build());
     }
 
-    /*
-     * Convert the connect timeout configured in the builder. If the timeout is null or less than or equal to zero a
-     * default timeout of 10 seconds will be used. Otherwise, the maximum of the configured timeout and one millisecond
-     * is used.
-     */
-    private static Duration convertConnectTimeout(Duration timeout) {
-        return convertTimeout(timeout, DEFAULT_CONNECT_TIMEOUT, true);
+    private static long getDefaultTimeout(String property, long defaultTimeoutMillis) {
+        String envTimeout = Configuration.getGlobalConfiguration().get(property);
+        if (CoreUtils.isNullOrEmpty(envTimeout)) {
+            return defaultTimeoutMillis;
+        }
+
+        try {
+            return Long.parseLong(envTimeout);
```

Comment:
We should also have bounds check to see if the configured value is not negative or 0.